website/content/en/preview/getting-started/migrating-from-cas/_index.md

```diff
@@ -0,0 +1,326 @@
+
+aws iam put-role-policy --role-name KarpenterController \
+    --policy-name KarpenterController \
+    --policy-document file://karpenter-controller-policy.json
+```
+
+## Add tags to subnets and security groups
+
+We need to add tags to our nodegroup subnets so Karpenter will know which subnets to use.
+
+```bash
+for NODEGROUP in $(aws eks list-nodegroups --cluster-name ${CLUSTER_NAME} \
+    --query 'nodegroups' --output text); do aws ec2 create-tags \
+        --tags "Key=karpenter.sh/discovery,Value=${CLUSTER_NAME}" \
+        --resources $(aws eks describe-nodegroup --cluster-name ${CLUSTER_NAME} \
+        --nodegroup-name $NODEGROUP --query 'nodegroup.subnets' --output text )
+done
+```
+
+Add tags to our node group.
+This command only tags the security groups for the first nodegroup in the cluster.
+If you have multiple nodegroups or multiple security groups you will need to decide which one Karpenter should use.
+
+```bash
+NODEGROUP=$(aws eks list-nodegroups --cluster-name ${CLUSTER_NAME} \
+    --query 'nodegroups[0]' --output text)
+
+LAUNCH_TEMPLATE=$(aws eks describe-nodegroup --cluster-name ${CLUSTER_NAME} \
+    --nodegroup-name ${NODEGROUP} --query 'nodegroup.launchTemplate.{id:id,version:version}' \
+    --output text | tr -s "\t" ",")
+
+SECURITY_GROUPS=$(aws ec2 describe-launch-template-versions \
+    --launch-template-id ${LAUNCH_TEMPLATE%,*} --versions ${LAUNCH_TEMPLATE#*,} \
+    --query 'LaunchTemplateVersions[0].LaunchTemplateData.SecurityGroupIds' \
+    --output text)
+
+aws ec2 create-tags \
+    --tags "Key=karpenter.sh/discovery,Value=${CLUSTER_NAME}" \
+    --resources ${SECURITY_GROUPS}
+```
+
+## Update aws-auth ConfigMap
+
+We need to allow nodes that are using the node IAM role we just created to join the cluter.
+To do that we have to modify the `aws-auth` ConfigMap in the cluster.
+
+```
+kubectl edit configmap aws-auth -n kube-system
+```
+
+You will need to add a section to the mapRoles that looks something like this.
+Replace the `${ACCOUNT_NUMBER}` variable with your account, but do not replace the `{{EC2PrivateDNSName}}`.
+```
+    - groups:
+      - system:bootstrappers
+      - system:nodes
+      rolearn: arn:aws:iam::${ACCOUNT_NUMBER}:role/KarpenterInstanceNodeRole
+      username: system:node:{{EC2PrivateDNSName}}
+```
+
+The full aws-auth configmap should have two groups.
+One for your Karpenter node role and one for your existing node group.
+
+## Deploy Karpenter
+
+We can now generate a full Karpenter deployment yaml from the helm chart.
+
+```bash
+KARPENTER_VERSION="v0.8.1"
+
+helm template --namespace karpenter \
+    karpenter karpenter/karpenter \
+    --set aws.defaultInstanceProfile=KarpenterInstanceNodeRole \
+    --set clusterEndpoint="${OIDC_ENDPOINT}" \
+    --set clusterName=${CLUSTER_NAME} \
+    --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"="arn:aws:iam::${ACCOUNT_NAME}:role/KarpenterController" \
+    --version ${KARPENTER_VERSION} > karpenter.yaml
+```
```

Comment:
Can you do this the same way we do:
https://github.com/aws/karpenter/blame/e7f4183921fa172f2afb5c126eed164fca19ca08/website/content/en/preview/getting-started/getting-started-with-eksctl/_index.md#L47

and https://github.com/aws/karpenter/blame/e7f4183921fa172f2afb5c126eed164fca19ca08/website/content/en/preview/getting-started/getting-started-with-eksctl/_index.md#L52
https://github.com/aws/karpenter/blob/e7f4183921fa172f2afb5c126eed164fca19ca08/website/content/en/preview/getting-started/getting-started-with-eksctl/scripts/step08-apply-helm-chart.sh

Reply:
I moved all of the essetial/runnable scripts to separate files. There are some commands that require the user to do things interactively (eg update aws-auth and modify generated helm chart)

Reply:
Do you know why after moving to script files the rendering of the page is broken?

https://deploy-preview-1642--karpenter-docs-prod.netlify.app/preview/getting-started/migrating-from-cas/

Reply:
Seems to be working, no?